pre_commit/languages/pygrep.py

```diff
@@ -75,11 +117,10 @@ def main(argv: Optional[Sequence[str]] = None) -> int:
     pattern = re.compile(args.pattern.encode(), flags)
 
     retv = 0
+
     for filename in args.filenames:
-        if args.multiline:
-            retv |= _process_filename_at_once(pattern, filename)
-        else:
-            retv |= _process_filename_by_line(pattern, filename)
+        process_fn = FNS[Choice(multiline=args.multiline, negate=args.negate)]
```

Comment:
this should be outside the loop, otherwise you're doing this work every iteration instead of just once